.github/workflows/codeql.yml

```diff
@@ -0,0 +1,83 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main" ]
```

Comment:
Do we need to mention the branch? As I understand we are delivering from `main` only, and older branches won't get any cherry picks?

Reply:
As discussed, normally cherry picks should generally end up in the main branch and we don't have any other delivery/protected branches.
This would work without mentioning branch, I just thought we could go with it explicitly.

Reply:
Explicit branch works as well! Thanks!